ui/pages/editor.tsx

```diff
@@ -43,6 +60,45 @@ export default function EditorPage({ code, fileName: initialFileName }) {
       });
   }
 
+  const fileOpenCallback = (filePath: string): (() => void) => {
+    return () => {
+      let url = path.join("/code", filePath);
+      fetch(url + window.location.search)
+        .then((resp) => resp.text())
+        .then((code) => {
+          setDisplayedCode(code);
```

Comment:
If you update the `name` state variable here via `setName` (also feel free to rename if that's confusing) then the editor should start picking up the desired language and syntax highlighting via the language check on line 102